R/mutate.R

```diff
@@ -142,9 +144,18 @@ mutate <- function(.data, ...) {
 }
 
 #' @export
-mutate.data.frame <- function(.data, ...) {
+mutate.data.frame <- function(.data, ..., .before = NULL, .after = NULL) {
   cols <- mutate_cols(.data, ...)
-  dplyr_col_modify(.data, cols)
+  out <- dplyr_col_modify(.data, cols)
+
+  if (!quo_is_null(enquo(.before)) || !quo_is_null(enquo(.after))) {
+    # Only change the order of new columns
+    new <- setdiff(names(cols), names(data))
+    relocate(out, any_of(new), .before = {{ .before }}, .after = {{ .after }})
```

Comment:
It looks like we can combine `.before` with `.remove = "used"` if we update `out` before applying the `.remove` logic.

```suggestion
    out <- relocate(out, any_of(new), .before = {{ .before }}, .after = {{ .after }})
    out
```

Reply:
Oh yeah, makes sense.